spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/server/app/SkipperServerApplication.java

```diff
@@ -13,24 +13,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.springframework.cloud.skipper.server;
+package org.springframework.cloud.skipper.server.app;
 
 import org.springframework.boot.SpringApplication;
-import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeployerAutoConfiguration;
 import org.springframework.cloud.deployer.spi.kubernetes.KubernetesAutoConfiguration;
 import org.springframework.cloud.deployer.spi.local.LocalDeployerAutoConfiguration;
-import org.springframework.data.map.repository.config.EnableMapRepositories;
+import org.springframework.cloud.skipper.server.EnableSkipperServer;
 
-@SpringBootApplication
-@EnableAutoConfiguration(exclude = { CloudFoundryDeployerAutoConfiguration.class,
-		LocalDeployerAutoConfiguration.class, KubernetesAutoConfiguration.class })
-@EnableMapRepositories
+@SpringBootApplication(exclude = { CloudFoundryDeployerAutoConfiguration.class, LocalDeployerAutoConfiguration.class,
```

Comment:
It might be better to exclude these autoconfiguration classes inside the config (maybe the application.yml) rather than the main application here. That way, a custom main application with `EnableSkipperServer` would still need not exclude these autoconfiguration classes explicitly.

Reply:
Nope, we can't. You can't exclude anything within a config itself as that will just create a mess. Exclude needs to happen on a top level component(SkipperServerApplication) or with a property(we disabled SessionAutoConfiguration) in scdf that way. I'm more interested to know why those auto-configs had to be excluded.

Reply:
we want to have control over creating the deployer configuration and that's why we disable these deployer auto configurations explicitly.

Reply:
we should improve their auto-config so that we wouldn't need to disable.

Reply:
yeah, I agree that the deployer auto configurations are flexible enough to configure enable/disable based on some conditions. In the meantime, I think we are OK having this (with just a note that every main application needs to disable these auto configurations explicitly).